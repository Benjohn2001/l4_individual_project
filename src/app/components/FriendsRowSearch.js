import React, { useState } from "react";
import PropTypes from "prop-types";
import { TouchableOpacity, Text, View, Image } from "react-native";
import { Feather } from "@expo/vector-icons";
import { getStorage, getDownloadURL, ref as stref } from "firebase/storage";

function FriendsRowSearch(props) {
  const [pic, setPic] = useState("");
  const { title, onPress, data } = props;

  React.useEffect(() => {
    if (data.val().profilePic !== undefined) {
      getDownloadURL(stref(getStorage(), data.val().profilePic)).then((url) => {
        setPic(url);
      });
    } else {
      setPic(
        Image.resolveAssetSource(require("../assets/defaultProfilePic.png")).uri
      );
    }
  }, []);

  return (
    <View className="pt-3 w-full items-center">
      {pic ? (
        <TouchableOpacity
          onPress={onPress}
          className="bg-secondaryPurple h-16 pt-1 w-11/12  rounded-2xl mx-1.5"
        >
          <View className="flex-row">
            <View className="ml-3">
              <Image className="w-14 h-14 rounded-full" source={{ uri: pic }} />
            </View>
            <Text className="text-black text-sm text-center ml-10 pt-4">
              {title}
            </Text>
            <View className="ml-auto mr-5 pt-4">
              <Feather name="arrow-right" size={20} color="black" />
            </View>
          </View>
        </TouchableOpacity>
      ) : (
        <></>
      )}
    </View>
  );
}

FriendsRowSearch.propTypes = {
  title: PropTypes.string.isRequired,
  onPress: PropTypes.func.isRequired,
  data: PropTypes.object,
};

export default FriendsRowSearch;
