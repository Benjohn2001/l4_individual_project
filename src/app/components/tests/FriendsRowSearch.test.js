import FriendsRowSearch from '../FriendsRowSearch';
import { render, screen, fireEvent } from '@testing-library/react-native';
import renderer from "react-test-renderer";

test('Check FriendsRowSearch renders correctly and onpress works',()=>{
    const data1 = {
        "bio": "Add a bio!",
        "email": "ts@gmail.com",
        "firstName": "Tara",
        "lastName": "Smith",
        "userName": "tdawg123"
      }
    const snap1 = { val: () => data1 };
    const onPressMock = jest.fn();
    render(
        <FriendsRowSearch 
            title="press"
            onPress={()=>{
                onPressMock()
            }}
            data={snap1}
        />,
      );
      fireEvent.press(screen.getByText('press'));
      expect(onPressMock)
      const domTree = renderer.create(
        <FriendsRowSearch 
            title="press"
            onPress={()=>{
                onPressMock()
            }}
            data={snap1}
        />
        ).toJSON();
      expect(domTree).toMatchSnapshot();
})