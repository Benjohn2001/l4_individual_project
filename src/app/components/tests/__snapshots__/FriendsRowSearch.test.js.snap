// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check FriendsRowSearch renders correctly and onpress works 1`] = `
<View
  style={
    Array [
      Object {
        "paddingTop": 12,
      },
      Object {
        "width": "100%",
      },
      Object {
        "alignItems": "center",
      },
    ]
  }
/>
`;
